[glass-effect] new package to render Glass effects on iOS (#39322)
expo/expo: docs/pages/versions/unversioned/sdk/glass-effect.mdx

---
title: GlassEffect
description: React components that render a glass effect using iOS's native UIVisualEffectView.
sourceCodeUrl: 'https://github.com/expo/expo/tree/main/packages/expo-glass-effect'
packageName: 'expo-glass-effect'

> platforms: ['ios']

Reviewer: ```suggestion
platforms: ['ios']
isNew: true
```

Let's add this to display the new badge for this page.

---

import APISection from '~/components/plugins/APISection';
import { APIInstallSection } from '~/components/plugins/InstallSection';
import { SnackInline } from '~/ui/components/Snippet';

> React components that render native iOS liquid glass effect using [UIVisualEffectView](https://developer.apple.com/documentation/uikit/uivisualeffectview). Supports customizable glass styles and tint color.
> 
> > **info** `GlassView` is only available on iOS 26+. It will fallback to regular View on unsupported platforms.

Reviewer: ```suggestion
> **important** `GlassView` is only available on iOS 26 and above. It will fallback to regular `View` on unsupported platforms.

React components that render native iOS liquid glass effect using [`UIVisualEffectView`](https://developer.apple.com/documentation/uikit/uivisualeffectview). Supports customizable glass styles and tint color.
```

Let's add the callout before the introduction.

## Installation

<APIInstallSection />

## Usage

> ### Glass view

Reviewer: ```suggestion
### `GlassView`
```

Let's use the component name here.

The `GlassView` component renders the native iOS glass effect. It supports different glass effect styles and can be customized with tint colors for various aesthetic needs.

<SnackInline label='Basic GlassView usage' dependencies={['expo-glass-effect']} platforms={['ios']}>

```jsx

> import React from 'react';

Reviewer: ```suggestion
```

Unused import statement. Not required in the example code.

export default function App() {
  return (
    <View style={styles.container}>
      <Image
        style={styles.backgroundImage}
        source={{
          uri: 'https://images.unsplash.com/photo-1506905925346-21bda4d32df4?w=400&h=400&fit=crop',
        }}
      />

      {/* Basic Glass View */}
      <GlassView style={styles.glassView} />

      {/* Glass View with clear style */}
      <GlassView style={styles.tintedGlassView} glassEffectStyle="clear" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  backgroundImage: {
    ...StyleSheet.absoluteFill,
    width: '100%',
    height: '100%',
  },
  glassView: {
    position: 'absolute',
    top: 100,
    left: 50,
    width: 200,
    height: 100,
    borderRadius: 12,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
  },
  tintedGlassView: {
    position: 'absolute',
    top: 250,
    left: 50,
    width: 200,
    height: 100,
    borderRadius: 12,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
  },
  text: {
    fontSize: 16,
    fontWeight: '600',
    color: '#000',
    textAlign: 'center',
  },
});
```

</SnackInline>

> ### Glass container

Reviewer: ```suggestion
### `GlassContainer`
```

> The `GlassContainer` component allows you to combine multiple glass views into a combined effect:

Reviewer: ```suggestion
The `GlassContainer` component allows you to combine multiple glass views into a combined effect.
```

<SnackInline label='GlassContainer example' dependencies={['expo-glass-effect']} platforms={['ios']}>

```jsx

> import React from 'react';

Reviewer: ```suggestion
```

    <View style={styles.container}>
      <Image
        style={styles.backgroundImage}
        source={{
          uri: 'https://images.unsplash.com/photo-1547036967-23d11aacaee0?w=400&h=600&fit=crop',
        }}
      />
      <GlassContainer spacing={10} style={styles.containerStyle}>
        <GlassView style={styles.glass1} isInteractive />
        <GlassView style={styles.glass2} />
        <GlassView style={styles.glass3} />
      </GlassContainer>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  backgroundImage: {
    ...StyleSheet.absoluteFillObject,
    width: '100%',
    height: '100%',
  },
  containerStyle: {
    position: 'absolute',
    top: 200,
    left: 50,
    width: 250,
    height: 100,
    flexDirection: 'row',
    alignItems: 'center',
    gap: 5,
  },
  glass1: {
    width: 60,
    height: 60,
    borderRadius: 30,
  },
  glass2: {
    width: 50,
    height: 50,
    borderRadius: 25,
  },
  glass3: {
    width: 40,
    height: 40,
    borderRadius: 20,
  },
});
```

</SnackInline>

## API

```js
import { GlassView, GlassContainer } from 'expo-glass-effect';
```

<APISection packageName="expo-glass-effect" />

> ### Known issues
> 
> The `isInteractive` prop can only be set once on mount and cannot be changed dynamically after the component has been rendered. If you need to toggle interactive behavior, you must remount the component with a different `key`.

Reviewer: ```suggestion
### Known issues

The `isInteractive` prop can only be set once on mount and cannot be changed dynamically after the component has been rendered. If you need to toggle interactive behavior, you must remount the component with a different `key`.
```

Let's move this section after Installation. This is a common pattern we follow in other SDK docs as well.
